docs/developer/architecture/core/uisettings-service.asciidoc

```diff
@@ -3,14 +3,68 @@
 
 NOTE: The UI settings service is available both server and client side.
 
-=== Server side usage
+=== Overview
+
+UI settings are configurable from the Advanced Settings page in Management and control the behavior of {kib}. `uiSettings` are stored in a `config` saved object and, as such, conform to the same conditions as other saved objects.
+
+There are several ways to configure an advanced setting:
+
+- Through the <<advanced-options, Advanced Settings UI>> (security restrictions apply)
+- Locked via kibana.yml's {kib-repo}blob/{branch}/src/core/server/ui_settings/ui_settings_defaults_client.ts[uiSettings.overrides.<key>]
+- Through the client-side `uiSettings` Service
+- Through the server-side `uiSettings` Service
```

Comment:
nit: In terms of structure, I'd consider treating these bullets as a sort of table of contents for the rest of the page, and make each entry an anchor link to a dedicated section on that topic. (You already have dedicated sections for client/server side, so it would mean making the overrides a dedicated section).

IMO a structure like this would help with the overall flow. Maybe something along the lines of:

---

### Overview
UI settings are configurable from the Advanced Settings page in Management and control the behavior of Kibana. uiSettings are stored in a config saved object and, as such, conform to the same conditions as other saved objects.

There are several ways to configure an advanced setting:

- [Through the Advanced Settings UI](#)
- [Locked via kibana.yml’s `uiSettings.overrides.<key>`](#)
- [Through the client-side uiSettings Service](#)
- [Through the server-side uiSettings Service](#)

Keep in mind that once you add a new advanced setting, you cannot change or remove it without [registering a migration in core](#).

### Configuration with Advanced Settings UI
The uiSettings service is the programmatic interface to Kibana's Advanced Settings UI. Kibana plugins use the service to extend Kibana UI Settings Management with custom settings for their plugin.

Configuration through the Advanced settings UI is restricted to users authorised to access the Advanced Settings page. Users who don’t have permissions to change these values default to using the settings configured to the space they are in. Config saved objects can be shared between spaces.

### Configuration with UI settings overrides
When a setting is configured as an override in kibana.yml, it will override any other value stored in the config saved object. If an override is misconfigured, it will fail config validation and will prevent Kibana from starting up. The override applies to Kibana as a whole for all spaces and users and the option will be disabled in the Advanced Settings page. We refer to these as "global" overrides.

### Configuration on the client side
blahblah

### Configuration on the server side
blahblahblah

### Migrations
blahblahblah

Reply:
Done